File: ASM2_API/ASM-Java5/src/main/java/com/dto/user/UserCreateRequest.java
package com.dto.user;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

import java.time.LocalDate;

public record UserCreateRequest(

        @NotBlank(message = "Email can not be blank")
        @Email(message = "Email is of incorrect format")
        String email,

        String username,
        String firstName,
        String lastName,
        LocalDate birthday,

        @NotBlank(message = "Password can not be blank")
        @Pattern(message = "Password must contain one: lowercase letter, uppercase letter, special char (!@#$%^&*_-), and be 8-32 chars long", regexp = "^(?=.*[A-Z])(?=.*[a-z])(?=.*[!@#$%^&*_-]).{8,32}$")
        String password,

        String phoneNumber,
        Integer roleId
) {
}

File: ASM2_API/ASM-Java5/src/main/kotlin/com/dto/user/UserCreateRequest.kt
package com.dto.user

import java.time.LocalDate

data class UserCreateRequest(
    val email: String,
    val username: String,
    val firstName: String?,
    val lastName: String?,
    val birthday: LocalDate?,
    val password: String,
    val phoneNumber: String?,
    val roleId: Int
)
